$NetBSD: patch-lib_fflush_c,v 1.3 2018/01/22 04:40:07 dbj Exp $

fflush.c:108:16: error: incompatible types when assigning to type '__off_t' from type 'fpos_t'
   fp_->_offset = u.f;
                ^

--- lib/fflush.c.orig	2012-01-29 01:22:33.000000000 +0900
+++ lib/fflush.c	2014-11-20 23:50:41.000000000 +0900
@@ -91,7 +91,7 @@
 update_fpos_cache (FILE *fp, off_t pos)
 {
 #if defined __sferror || defined __DragonFly__ /* FreeBSD, NetBSD, OpenBSD, DragonFly, MacOS X, Cygwin */
-# if defined __CYGWIN__
+# if defined(__CYGWIN__) || defined(__FreeBSD__) || defined(__APPLE__)
   /* fp_->_offset is typed as an integer.  */
   fp_->_offset = pos;
 # else
@@ -105,7 +105,7 @@
       off_t o;
     } u;
   u.o = pos;
-  fp_->_offset = u.f;
+  fp_->_offset = u.f._pos;
 # endif
   fp_->_flags |= __SOFF;
 #endif
